src/inputs/labeled-inputs.scss

```diff
@@ -28,177 +66,214 @@
       cursor: not-allowed;
     }
   }
+  // #endregion
 
-  > .iui-label {
-    margin-bottom: $iui-component-offset;
-    font-weight: $iui-font-weight-semibold;
-
-    &.iui-required {
-      &::after {
-        content: ' *';
-        @include themed {
-          color: t(iui-color-foreground-negative);
-        }
-      }
-    }
+  // Appropriate spacing even if no text label is given
+  .iui-checkbox,
+  .iui-radio {
+    min-height: $iui-line-height;
   }
 
-  > .iui-message {
-    display: flex;
-    width: fit-content;
-    align-items: center;
-    font-size: $iui-font-size-small;
-    line-height: $iui-line-height;
-    margin-top: $iui-component-offset;
-    @include themed {
-      color: t(iui-text-color-muted);
-    }
+  &.iui-inline-label {
+    grid-template: 'label inputs icon' / auto 1fr auto;
 
-    svg {
-      width: $iui-icons-default;
-      height: $iui-icons-default;
-      display: flex; // needed for vertical alignment of webcomponents
-      margin-right: $iui-xs;
-      @include themed {
-        fill: t(iui-icons-color);
+    // IE fix icon position for inline checkboxes/radios group
+    @at-root {
+      div#{&} {
+        // stylelint-disable -- fix position in IE
+        -ms-grid-columns: auto min-content auto;
+        // stylelint-enable
       }
     }
 
-    > .iui-borderless {
-      border-top-left-radius: 0;
-      border-bottom-left-radius: 0;
-
-      svg {
-        @include themed {
-          fill: t(iui-icons-color-actionable);
-        }
-      }
+    .iui-input,
+    .iui-textarea,
+    .iui-input-group,
+    .iui-select {
+      // stylelint-disable -- fix position in IE
+      -ms-grid-row: 1;
+      -ms-grid-column: 2;
+      // stylelint-enable
     }
 
-    a {
-      user-select: none;
+    .iui-input-icon {
+      // stylelint-disable -- fix position in IE
+      -ms-grid-row: 1;
+      -ms-grid-column: 3;
+      // stylelint-enable
     }
-  }
-
-  // Appropriate spacing even if no text label is given
-  > .iui-checkbox,
-  > .iui-radio {
-    min-height: $iui-line-height;
-  }
 
-  &.iui-inline {
-    flex-direction: row;
-    align-items: center;
+    &.iui-with-message {
+      grid-template:
+        'label inputs inputs'
+        '. icon message' / auto auto 1fr;
 
-    > .iui-label {
-      flex-shrink: 0;
-      margin: 0 $iui-m 0 0;
+      &:not(.iui-inline-icon) .iui-input-icon {
+        // stylelint-disable -- fix position in IE
+        -ms-grid-row: 2;
+        -ms-grid-column: 2;
+        // stylelint-enable
+      }
 
-      &.iui-required {
-        margin-right: $iui-xs * 1.5;
+      .iui-message {
+        // stylelint-disable -- fix position in IE
+        -ms-grid-row: 2;
+        -ms-grid-column: 3;
+        // stylelint-enable
       }
     }
 
-    > .iui-checkbox:not(:last-child),
-    > .iui-radio:not(:last-child),
-    > .iui-toggle-switch:not(:last-child) {
-      margin-right: $iui-m;
-    }
+    > .iui-input-group {
+      display: flex;
+      gap: $iui-m;
```

Comment:
Are we ok with using `gap`?

Reply:
Ah wait, I see `@supports (gap: $iui-m)` a couple lines down which I'm assuming is the IE solution.

Reply:
Oh wait, I think I see what is happening down below:
1. It tries to apply gap.
2. If a browser doesn't support gap, it also applies margin-right to try to replicate gap.
3. If a browser _does_ support gap, using `@supports (gap)` we then remove the margin-right & only rely on gap.
Clever.